src/Bundle/DependencyInjection/Compiler/RegisterFiltersPass.php

```diff
@@ -44,6 +44,10 @@ public function process(ContainerBuilder $container): void
             }
 
             foreach ($attributes as $attribute) {
+                if (isset($attribute['type'], $attribute['form_type'])) {
```

Comment:
Isn't it?
```suggestion
                if (!isset($attribute['type'], $attribute['form_type'])) {
```

Reply:
No, I check the presence of a type and form_type attribute to avoid reconfiguring a filter that already has with Attribute. To keep a compatibility with the previous Sylius versions, we need to keep the autowire of all FilterInterface which leads to this issue.

Reply:
In the case of Attribute and FilterInterace, the $attributes will look like this :

```php
[
  'type' => 'foo',
  'form_type' => FormType::Class
],
[]
```

The first iteration can be skipped to the second part of the array but it means you register your filter twice which is never wanted AFAIK. (@loic425 confirmed that too)

I'm open to suggestion but the BC thing is annoying in this case.

Reply:
Ha right, got it ;)